src/devices/machine/68230pit.cpp

```diff
@@ -214,16 +226,32 @@ void pit68230_device::device_timer (emu_timer &timer, device_timer_id id, int32_
 
 void pit68230_device::h1_set (uint8_t state)
 {
-	LOG(("%s %s %d @ m_psr %2x => ",tag(), FUNCNAME, state, m_psr));
 	if (state) m_psr |= 1; else m_psr &= ~1;
-	LOG(("%02x %lld\n", m_psr, machine ().firstcpu->total_cycles ()));
 }
 
-void pit68230_device::portb_setbit (uint8_t bit, uint8_t state)
+void pit68230_device::portb_setbit(uint8_t bit, uint8_t state)
 {
-	LOG(("%s %s %d/%d @ m_pbdr %2x => ", tag(), FUNCNAME, bit, state, m_pbdr));
 	if (state) m_pbdr |= (1 << bit); else m_pbdr &= ~(1 << bit);
-	LOG(("%02x %lld\n", m_pbdr, machine ().firstcpu->total_cycles ()));
+}
+
+// Bit updaters will make it possible to move registers to private data
+// TODO: Make sure to only update input bits and that the port is in the right alternate mode
+void pit68230_device::pa_update_bit(uint8_t bit, uint8_t state){ if (state) m_padr |= (1 << bit); else m_padr &= ~(1 << bit); }
+void pit68230_device::pb_update_bit(uint8_t bit, uint8_t state){ if (state) m_pbdr |= (1 << bit); else m_pbdr &= ~(1 << bit); }
+void pit68230_device::pc_update_bit(uint8_t bit, uint8_t state){ if (state) m_pcdr |= (1 << bit); else m_pcdr &= ~(1 << bit); }
+
+void pit68230_device::update_tin()
+{ 
+	static uint32_t counter = 0;
```

Comment:
This can't possibly be right - how can there be one of these for all instances of the device?  Function statics are also a problem for when we want to be able to run multiple machines in the future.